src/lib/SpecialFileDirGen.cc

```diff
@@ -4,19 +4,20 @@
 
 namespace simeng {
 
-SpecialFileDirGen::SpecialFileDirGen(YAML::Node config) {
+SpecialFileDirGen::SpecialFileDirGen() {
+  ryml::ConstNodeRef config = config::SimInfo::getConfig();
   // Import all values from config file
-  core_count = config["CPU-Info"]["Core-Count"].as<uint64_t>();
-  socket_count = config["CPU-Info"]["Socket-Count"].as<uint64_t>();
-  smt = config["CPU-Info"]["SMT"].as<uint64_t>();
-  bogoMIPS = config["CPU-Info"]["BogoMIPS"].as<float>();
-  features = config["CPU-Info"]["Features"].as<std::string>();
-  cpu_implementer = config["CPU-Info"]["CPU-Implementer"].as<std::string>();
-  cpu_architecture = config["CPU-Info"]["CPU-Architecture"].as<uint64_t>();
-  cpu_variant = config["CPU-Info"]["CPU-Variant"].as<std::string>();
-  cpu_part = config["CPU-Info"]["CPU-Part"].as<std::string>();
-  cpu_revision = config["CPU-Info"]["CPU-Revision"].as<uint64_t>();
-  package_count = config["CPU-Info"]["Package-Count"].as<uint64_t>();
+  config["CPU-Info"]["Core-Count"] >> core_count;
+  config["CPU-Info"]["Socket-Count"] >> socket_count;
+  config["CPU-Info"]["SMT"] >> smt;
+  config["CPU-Info"]["BogoMIPS"] >> bogoMIPS;
+  config["CPU-Info"]["Features"] >> features;
+  config["CPU-Info"]["CPU-Implementer"] >> cpu_implementer;
+  config["CPU-Info"]["CPU-Architecture"] >> cpu_architecture;
+  config["CPU-Info"]["CPU-Variant"] >> cpu_variant;
+  config["CPU-Info"]["CPU-Part"] >> cpu_part;
+  config["CPU-Info"]["CPU-Revision"] >> cpu_revision;
+  config["CPU-Info"]["Package-Count"] >> package_count;
```

Comment:
No use of SimInfo's `getValue` function which allows for a more conventional way of setting values (more readable to do `var_a = SimInfo::getValue<int>(config[...]);`

Reply:
This is a re-occuring issue throughout - please apply this throughout the project

Reply:
Config getter functionality has been changed to new custom ryml `as` function as discussed offline.